### PR TITLE
enhance(unstruct): add list support in unstruct package

### PR DESCRIPTION
**Pull request by ashishranjan738**
Signed-off-by: Ashish Ranjan <ashish.ranjan@mayadata.io>

**What this PR does / why we need it**:
This commit add list support for unstruct package which will help in operating over list of unstruct object

**Special notes for your reviewer**:

**Checklist:**
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [x] Commit has integration tests

### CODE REVIEW

File: pkg/unstruct/v1alpha2/unstruct.go [hunk 1]
```diff
@@ -27,9 +29,9 @@ type unstruct struct {
 	object *unstructured.Unstructured
 }
 
-// ToUnstructured converts unstruct object
+// GetUnstructured converts unstruct object
```
[the remark this answers is not in the document]
Comment (ashishranjan738) [reply pkg/unstruct/v1alpha2/unstruct.go]: done

File: pkg/unstruct/v1alpha2/unstruct.go [hunk 2]
```diff
@@ -78,3 +80,47 @@ func (b *builder) Build() (*unstruct, error) {
 	}
 	return b.unstruct, nil
 }
+
+// listBuilder enables building a list
+// of an unstruct instance
+type listBuilder struct {
+	items []*unstruct
```
[the remark this answers is not in the document]
Comment (ashishranjan738) [reply pkg/unstruct/v1alpha2/unstruct.go]: done

File: pkg/unstruct/v1alpha2/unstruct.go [hunk 3]
```diff
@@ -78,3 +80,47 @@ func (b *builder) Build() (*unstruct, error) {
...
+	errs  []error
+}
+
+// ListUnstructBuilderForYaml returns a mew instance of
+// list unstruct builder by making use of the provided YAMLs
+func ListUnstructBuilderForYaml(docs string) *listBuilder {
+	lb := &listBuilder{}
+	sepYamlfiles := strings.Split(docs, "---")
```
[the remark this answers is not in the document]
Comment (ashishranjan738) [reply pkg/unstruct/v1alpha2/unstruct.go]: done

File: pkg/unstruct/v1alpha2/unstruct.go [hunk 4]
```diff
@@ -78,3 +80,47 @@ func (b *builder) Build() (*unstruct, error) {
...
+	for _, f := range sepYamlfiles {
+		if f == "\n" || f == "" {
+			// ignore empty cases
+			continue
+		}
+		f = strings.TrimSpace(f)
+		a, err := UnstructBuilderForYaml(f).Build()
+		if err != nil {
+			lb.errs = append(lb.errs, err)
```
[the remark this answers is not in the document]
Comment (ashishranjan738) [reply pkg/unstruct/v1alpha2/unstruct.go]: done

File: pkg/unstruct/v1alpha2/unstruct.go [hunk 5]
```diff
@@ -78,3 +80,47 @@ func (b *builder) Build() (*unstruct, error) {
...
+		}
+		lb.items = append(lb.items, a)
+	}
+	return lb
+}
+
+// ListUnstructBuilderForObject returns a mew instance of
```
[the remark this answers is not in the document]
Comment (ashishranjan738) [reply pkg/unstruct/v1alpha2/unstruct.go]: done

File: pkg/unstruct/v1alpha2/unstruct.go [hunk 6]
```diff
@@ -78,3 +80,47 @@ func (b *builder) Build() (*unstruct, error) {
...
+	for _, obj := range objs {
+		lb.items = append(lb.items, &unstruct{obj})
+	}
+	return lb
+}
+
+// Build returns the list of unstruct objects created by
+// the builder
+func (l *listBuilder) Build() ([]*unstruct, error) {
+	return l.items, errors.Errorf("%v", l.errs)
```
[the remark this answers is not in the document]
Comment (ashishranjan738) [reply pkg/unstruct/v1alpha2/unstruct.go]: done

File: pkg/unstruct/v1alpha2/unstruct_test.go [hunk 7]
```diff
@@ -104,3 +120,48 @@ func TestUnstructBuilderForYamlBuild(t *testing.T) {
 		})
 	}
 }
+
+func TestListUnstructBuilderForYaml(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (ashishranjan738) [reply pkg/unstruct/v1alpha2/unstruct_test.go]: done

File: pkg/unstruct/v1alpha2/unstruct.go [hunk 8]
```diff
@@ -78,3 +80,57 @@ func (b *builder) Build() (*unstruct, error) {
...
+// of an unstruct instance
+type listBuilder struct {
+	list unstructList
+	errs []error
+}
+
+// ListUnstructBuilderForYamls returns a mew instance of
+// list unstruct builder by making use of the provided YAMLs
+func ListUnstructBuilderForYamls(docs string) *listBuilder {
+	lb := &listBuilder{}
```
Comment (prateekpandey14) [inline pkg/unstruct/v1alpha2/unstruct.go]: IMO `ListBuilderForYamls` will be better name as package name is `unstruct` itself.. make the linter happy.

File: pkg/unstruct/v1alpha2/unstruct.go [hunk 9]
```diff
@@ -78,3 +80,57 @@ func (b *builder) Build() (*unstruct, error) {
...
+		lb.list.items = append(lb.list.items, a)
+	}
+	return lb
+}
+
+// ListUnstructBuilderForObjects returns a mew instance of
+// list unstruct builder by making use of the provided
+// unstructured object
+func ListUnstructBuilderForObjects(objs ...*unstructured.Unstructured) *listBuilder {
+	lb := &listBuilder{}
```
Comment (prateekpandey14) [inline pkg/unstruct/v1alpha2/unstruct.go]: Same as above comment,  ` ListBuilderForObjects`

File: pkg/unstruct/v1alpha2/unstruct.go [hunk 10]
```diff
@@ -78,3 +80,47 @@ func (b *builder) Build() (*unstruct, error) {
...
+// listBuilder enables building a list
+// of an unstruct instance
+type listBuilder struct {
+	items []*unstruct
+	errs  []error
+}
+
+// ListUnstructBuilderForYaml returns a mew instance of
+// list unstruct builder by making use of the provided YAMLs
+func ListUnstructBuilderForYaml(docs string) *listBuilder {
```
[the remark this answers is not in the document]
Comment (ashishranjan738) [reply pkg/unstruct/v1alpha2/unstruct.go]: done

File: pkg/unstruct/v1alpha2/unstruct.go (hunk 8 above)
Comment (ashishranjan738) [reply pkg/unstruct/v1alpha2/unstruct.go]: correct @prateekpandey14

File: pkg/unstruct/v1alpha2/unstruct.go (hunk 9 above)
Comment (ashishranjan738) [reply pkg/unstruct/v1alpha2/unstruct.go]: correct @prateekpandey14

**Pull request merged**